src/main/java/at/hannibal2/skyhanni/features/misc/customscoreboard/ScoreboardEvents.kt

```diff
@@ -0,0 +1,499 @@
+}
+
+private fun getFlightDurationShowWhen(): Boolean {
+    return SbPattern.flightDurationPattern.anyMatches(getSbLines())
+}
+
+private fun getWinterLines() = buildList {
+    addIfNotNull(getSbLines().firstOrNull { SbPattern.winterEventStartPattern.matches(it) })
+    addIfNotNull(getSbLines().firstOrNull { SbPattern.winterNextWavePattern.matches(it) && !it.endsWith("Soon!") })
+    addIfNotNull(getSbLines().firstOrNull { SbPattern.winterWavePattern.matches(it) })
+    addIfNotNull(getSbLines().firstOrNull { SbPattern.winterMagmaLeftPattern.matches(it) })
+    addIfNotNull(getSbLines().firstOrNull { SbPattern.winterTotalDmgPattern.matches(it) })
+    addIfNotNull(getSbLines().firstOrNull { SbPattern.winterCubeDmgPattern.matches(it) })
+}
+
+private fun getWinterShowWhen(): Boolean {
+    return getSbLines().any { ScoreboardPattern.winterEventStartPattern.matches(it) }
+        || getSbLines().any { ScoreboardPattern.winterNextWavePattern.matches(it) && !it.endsWith("Soon!") }
+        || getSbLines().any { ScoreboardPattern.winterWavePattern.matches(it) }
+}
+
+private fun getSpookyLines() = buildList {
+    addIfNotNull(getSbLines().firstOrNull { SbPattern.spookyPattern.matches(it) }) // Time
+    addIfNotNull("§7Your Candy: ")
+    addIfNotNull(
+        CustomScoreboardUtils.getTablistFooter()
+            .split("\n")
+            .firstOrNull { it.startsWith("§7Your Candy:") }
+            ?.removePrefix("§7Your Candy:") ?: "§cCandy not found"
+    ) // Candy
+}
+
+private fun getSpookyShowWhen(): Boolean {
+    return getSbLines().any { ScoreboardPattern.spookyPattern.matches(it) }
+}
+
+private fun getMarinaLines(): List<String> {
+    return listOf(
+        "§bFishing Festival: " + TabListData.getTabList().nextAfter("§e§lEvent: §r§bFishing Festival")
+            ?.removePrefix(" Ends In: ")
+    )
+}
+
+private fun getMarinaShowWhen(): Boolean {
+    return TabListData.getTabList()
+        .any { it.startsWith("§e§lEvent: §r§bFishing Festival") } && TabListData.getTabList()
+        .nextAfter("§e§lEvent: §r§bFishing Festival")?.startsWith(" Ends In: ") == true
+}
+
+private fun getBroodmotherLines(): List<String> {
+    return listOf(getSbLines().first { SbPattern.broodmotherPattern.matches(it) })
+}
+
+private fun getBroodmotherShowWhen(): Boolean {
+    return getSbLines().any { SbPattern.broodmotherPattern.matches(it) }
+}
+
+private fun getNewYearLines(): List<String> {
+    return listOf(getSbLines().first { SbPattern.newYearPattern.matches(it) })
+}
+
+private fun getNewYearShowWhen(): Boolean {
+    return getSbLines().any { SbPattern.newYearPattern.matches(it) }
+}
+
+private fun getOringoLines(): List<String> {
+    return listOf(getSbLines().first { SbPattern.travelingZooPattern.matches(it) })
+}
+
+private fun getOringoShowWhen(): Boolean {
+    return getSbLines().any { SbPattern.travelingZooPattern.matches(it) }
+}
+
+private fun getMiningEventsLines() = buildList {
```

Comment:
Idk about the first uses, since the may throw

Reply:
Tried at didnt seem to throw

Reply:
only because it didn't throw once does not mean it won't throw in the future.